src/__internal__/utils/helpers/tags/tags.ts

```diff
@@ -1,12 +1,21 @@
+interface Props {
+  "data-element"?: string;
+  "data-role"?: string;
+}
+
+interface TagProps extends Props {
+  "data-component": string;
+}
+
 /**
  * Builds props object containing top level data tags
  *
  * @param {String} name of component
  * @param {Object} component props
- * @return {Object} dataTagProps
+ * @return {Object} dataProps
  */
-function tagComponent(component, props) {
-  const tagProps = {
+function tagComponent(component: string, props: Props): TagProps {
```

Comment:
```suggestion
function tagComponent(componentName: string, dataProps: DataProps): TagProps {
```

Reply:
I don't think we should rename the second arg to `dataProps` as it will contain a mix of all props passed as `rest` to a component